gammapy/cube/mapmaker.py

```diff
@@ -0,0 +1,143 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+import logging
+from ..maps import WcsNDMap
+from astropy.nddata.utils import PartialOverlapError
+from .new import *
+
+__all__ =['MapMaker']
+class MapMaker(object):
+    """Make all basic maps from observations.
+
+    Parameters
+    ----------
+
+    ref_geom_ereco : `~gammapy.maps.WcsGeom`
+        Reference image geometry for the counts and background maps in reco energy
+    ref_geom_etrue : `~gammapy.maps.WcsGeom`
+
+    It is assumed that ref_geom_ereco and ref_geom_etrue have the same wcs but can have different
+    energy binning.
+
+    TODO: 1. Can they have different WCS and we do a reproject?
```

Comment:
@AtreyeeS - This `MapMaker` docstring is no longer properly formatted. If you run `python setup.py build_docs` locally and look at the HTML docs you will see that. If you want to keep these TODOs are reminder issues, please put them as code comments.

You can assume that all the WCS are the same, users should just reproject before using the MapMaker if needed.
We also shouldnt need two geom to specify this, maybe just take an extra energy Quantity vector, default to None and in that case make the two the same?